ex03.exs

```diff
@@ -51,12 +51,17 @@ defmodule Ex03 do
   """
 
   def pmap(collection, process_count, function) do
-    # your code goes here
-    # I'm hoping to see a simple pipeline as the body of this function...
+    chunk_val = div(Enum.count(collection),process_count)
+    chunks = Enum.chunk_every(collection, chunk_val)
+    pids = Enum.map(chunks, fn chunk -> spawn(Ex03, :chunk_process, [chunk, function, self()]) end)
+    result_list = Enum.map(pids, fn pid -> (receive do {^pid, value} -> value end) end)
+    List.flatten(result_list)
```

Comment:
This was so close to being a pipeline:

~~~ elixir
    collection
    |> Enum.chunk_every(chunk_val)
    |> Enum.map(fn chunk -> spawn(Ex03, :chunk_process, [chunk, function, self()]) end)
    |> Enum.map(fn pid -> (receive do {^pid, value} -> value end) end)
    |> List.flatten()
~~~

I'm deducting 2 here just because I did ask for one...

# -2

No deduction, but you could have used `Enum.flat_map` and saved having the `List.flatten` call.